src/main/java/org/example/EchoMessageBus.java

```diff
@@ -0,0 +1,44 @@
+package org.example;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+public class EchoMessageBus extends Thread {
+    private final Socket clientSocket;
+    private final InputOutput io;
+
+    public EchoMessageBus(Socket socket) {
+        this.clientSocket = socket;
+        io = new InputOutput();
+    }
+
+    public void run() {
+        try {
+            System.out.println("[CLIENT CONNECTED]");
+            BufferedReader in = io.getSocketInputStream(clientSocket);
+            PrintWriter out = io.sendSocketOutputStream(clientSocket);
+            String userInput;
+            while((userInput = io.readClientInputStream(in)) != null) {
```

Comment:
Seems like this class is a going off from our original design. But not too far. From our original design this class is a `ClientConnection` and since we are handling a single connection there is no `MessageBus` for now. Currently, this method seems to be doing two things - establishes the connection, then listens it and echos every input message back. Can we extract these three responsibilities from that method? Can for instance `BufferedReader` and `PrintWriter` be initialized in the constructor of this method? Can the listening while loop be extracted one level higher, so EchoServer is responsible for listening and connection only for receiving and publishing messages?

Reply:
Resolved while pairing